contract/src/msgs/data_requests/execute/post_request.rs

```diff
@@ -7,13 +7,25 @@ use crate::{
         consts::{MAX_REPLICATION_FACTOR, MIN_EXEC_GAS_LIMIT, MIN_GAS_PRICE, MIN_TALLY_GAS_LIMIT},
         state::DR_CONFIG,
     },
-    state::TOKEN,
+    state::{DR_POOL_DRAIN_TARGET, TOKEN},
     utils::get_attached_funds,
 };
 
 impl ExecuteHandler for execute::post_request::Execute {
     /// Posts a data request to the pool
     fn execute(self, deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response, ContractError> {
+        let dr_config = DR_CONFIG.load(deps.storage)?;
+
+        let target = DR_POOL_DRAIN_TARGET.load(deps.storage)?;
+        if target > 0 {
+            let max_blocks_until_expiration: u64 = dr_config.commit_timeout_in_blocks.get() as u64
+                + dr_config.reveal_timeout_in_blocks.get() as u64
+                + dr_config.backup_delay_in_blocks.get() as u64;
```

Comment:
Why do we need to account for the backup delay? 

Practically speaking I think overshooting by a few blocks is good, but I was wondering if there's a reason to make it equal to the backup delay :D

Reply:
This is incorrect.. I misunderstood the meaning of backup delay. Will remove it.

I agree slight overshooting is good. Maybe I can just add 5?